server/index.js

```diff
@@ -101,6 +105,42 @@ app.post('/api/menus', upload.single('image'), (req, res) => {
     .catch(() => res.status(500).json({ error: 'An unexpected error occured.' }));
 });
 
+app.put('/api/menus/:itemId', (req, res, next) => {
+  const itemId = req.params.itemId;
+  const item = req.body.item;
+  let cost = req.body.cost;
+  let salePrice = req.body.salePrice;
+
+  if (!checkValidity(itemId) || !item || !cost || !salePrice) {
+    next(new ClientError('Sorry, your requested information is invalid.', 400));
+  }
+
+  cost = parseFloat(cost).toFixed(2);
+  salePrice = parseFloat(salePrice).toFixed(2);
+  const url = req.body.image ? '/images/' + req.body.image : null;
+
+  const paramDb = [item, cost, salePrice, url, itemId];
+  const sql = `
+    update "menus"
+       set "item" = $1,
+           "cost" = $2,
+           "salePrice" = $3,
+           "imageUrl" = $4
+     where "itemId" = $5
+           returning *
+   `;
+
+  db.query(sql, paramDb)
+    .then(result => {
+      if (result.rows[0] === undefined) {
+        next(new ClientError('Requested itemId may not exist in the database. Check your data agin.', 404));
+      } else {
+        res.status(200).json(result.rows[0]);
+      }
+    })
+    .catch(() => res.status(500).json({ error: 'Database query failed' }));
```

Comment:
I think all the error handler method  gives a 500 for everything that isn't a ClientError anyway.

Reply:
When I test like this, /api/menus/35 which doesn't exist, I got 404 error.
I'm out of home, so if you can, you may test it,  or let's test all on the next week.